sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1062,8 +1069,10 @@ public Instant getWatermark() {
         return initialWatermark;
       }
 
+      //watermark increases as current_timestamp, in case of latency,
+      //developers need to either provide a customized watermarkFn, or tune Triggers.
       return source.spec.getWatermarkFn() != null
-          ? source.spec.getWatermarkFn().apply(curRecord) : curTimestamp;
```

Comment:
Agree with **1)** , 
> by default both `event_timestamp` and `watermark` as processing timestamps.

Developers can combine `event_timestamp` and `watermark` if required.